core/modules/image/image.field.inc

```diff
@@ -437,6 +437,11 @@ function image_field_widget_form(&$form, &$form_state, $field, $instance, $langc
 
     // If not using custom extension validation, ensure this is an image.
     $supported_extensions = array('png', 'gif', 'jpg', 'jpeg');
+    if (defined('IMAGETYPE_WEBP')) {
```

Comment:
Best practice might be to call the hook itself in this module instead of like this. Something like:

```
function image_image_supported_extensions_alter(&$supported_extensions) {
  if (defined('IMAGETYPE_WEBP')) {
    // The current php/gd version supports webp.
    $supported_extensions[] = 'webp';
  }
}
```

Reply:
This might make sense being moved to `system` module, since it declares the GD tookit in `system_image_toolkits()`. Really we should check if GD is the set toolkit as well, because the constant `IMAGETYPE_WEBP` could exist but it might not matter if GD is not the default toolkit.

Reply:
Makes sense. :+1: